File: src/main/scala/com/abaddon83/cardsGames/burracoGames/domainModels/burracoGames/initialised/playerInGames/Burraco.scala
package com.abaddon83.cardsGames.burracoGames.domainModels.burracoGames.initialised.playerInGames

import com.abaddon83.cardsGames.burracoGames.domainModels.BurracoId
import com.abaddon83.cardsGames.shares.decks.Card

trait Burraco {
  protected val burracoId: BurracoId
  protected val cards: List[Card]

  def getBurracoId(): BurracoId = {
    burracoId
  }

  def showCards(): List[Card] = cards

  def isBurraco(): Boolean = {
    if(cards.size < 7) {
      false
    }else{
      true
    }
  }

}

File: src/main/kotlin/com/abaddon83/burracoGame/shared/burracos/Burraco.kt
package com.abaddon83.burracoGame.shared.burracos

import com.abaddon83.burracoGame.shared.decks.Card
import com.abaddon83.utils.ddd.Entity


abstract class Burraco : Entity<BurracoIdentity>() {

    protected abstract val cards: List<Card>

    fun showCards(): List<Card> = cards

    fun isBurraco(): Boolean = cards.size >= 7

}